qbicsoftware/offer-manager-2-portlet
# Implement layout changes in offer PDF

**Description of changes**
Addresses #613 by adapting the dynamically changed layout to the html template. 

1. Introduce acronyms for the individual productCategories.
2. Add the acronyms to the table title and the net price listing in the subtotal footer of each productCategory table.  
3. Removes the overhead and total cost in the subtotal footer of each productCategory table. 
4. Rewrite final total pricing footer to match html template. 
5. Adapt the spacing calculation for each page hosting a productCategory table to avoid overflow into the fixed page footer.
6. Introduce `ItemPrintOut.pageBreak` method which is used to dynamically set pagebreaks when a page has no spacing left. 
7. Introduce `ItemPrintout.createNewTable` method which is used to create the scaffolding to which new productitems can be added.
8. Adapt `ItemPrintout.tableHeader `method to match html template. 

**Technical details**
The most critical change in this PR addresses the calculated spacing for the individual productCategory tables to avoid an overflow of the productitem table into the fixed footer. 
Currently this is done by incrementing the counter `productItemsCounter` for each element added to a page. 
This includes productItems, TableTitle and SubtableFooter since they all take up space on the page.   
The main problem is that this solution doesn't account for variations in the productItem descriptions length, which could lead to possible overflows into the fixed footer if the `maxTableItems` is set to high.

**How to test**
Compare the HTML Template in the `src/main/resources/offer-template/offer.html` with the dynamically created PDF of an offer. 
It's of critical importance that the tables don't overflow into the footer.
Therefore It would make sense to check if the spacing works for offers with a varied amount of productitems. 
Here is an example list of OfferIds of such offers.
`O_greiner_wydi_1`
`O_greiner_fudv_3`
`O_greiner_fudv_1`
`O_doctor_tdli_1`
